scipy/interpolate/fitpack/fpregr.f

```diff
@@ -267,6 +267,7 @@ recursive subroutine fpregr(iopt,x,mx,y,my,z,mz,xb,xe,yb,ye,
           call fpknot(x,mx,tx,nx,fpintx,nrdatx,nrintx,nxest,1)
 c  test whether we cannot further increase the number of knots in the
 c  x-direction.
+          if(nx.eq.nmaxx) go to 10
```

Comment:
So this line is: if we have nmaxx knots, we need to make sure that the x-knots, tx, satisfies the not-a-knot boundary condition. But the y-knots, ty, can be whatever they are.
However IIUC  the `goto 10` makes both x- and y- knots to be not-a-knot.

So I believe you need to make sure that `tx` is indeed not-a-knot and `ty` is what it was at line 269.

Reply:
Makes sense. 100%. Let me first address this review and then other ones.

Reply:
Made the changes. Please verify the logic. Basically we just note that if we are coming to `10` label from `if(nx.eq.nmaxx) go to 10` (by checking `onlyxknots`) and skip `ty` re-computation in this case.

Reply:
Please double-check the `onlyxknots` logic. After `goto 10`, what is there to avoid recomputing `ty`?

Reply:
looks better now, thanks